pmd-ui/src/main/java/net/sourceforge/pmd/util/fxdesigner/XPathPanelController.java

```diff
@@ -127,8 +128,17 @@ public void initialize(URL location, ResourceBundle resources) {
 
         xpathExpressionArea.plainTextChanges()
                            .map(m -> {
-                               System.out.println(m.getInsertionEnd() + " ins");
-                               System.out.println(m.getRemovalEnd() + " rem");
+                               if (m.getInsertionEnd() > m.getRemovalEnd()) {
```

Comment:
When just watching the screen this *appears* to work but it fires many exceptions (because of index out of bounds) and I don't think it's good enough. I don't see a good reason to use fields when you can hand parameters to your autocomplete method using a tuple.